README.md

```diff
@@ -1,2 +1,90 @@
 # elmer-adapter
 **experimental** preCICE-adapter for the open source multiphysical simulation software Elmer FEM
+
+# TODO
+* remove personal remind quotes
+* add result image for example problems
+* Complete required software and installation instructions
+
+# Required Software
+* libprecice2_2.2.1(libprecice2_2.2.1_bionic.deb), make sure to install compatible version with operating system, here bionic is used because i have ubuntu 18.04
+
+
+# Installation Instructions
+* preCICE
+    * install debian package(can also compile source code but package is easier)
+    * GO TO https://github.com/precice/precice/releases/tag/v2.2.1 and download libprecice2_2.2.1_bionic.deb, (remind quote, saved in precice software)
+    * GO TO the location of the downloaded debian package,type sudo dpkg -i libprecice2_2.2.1_bionic.deb
+    * To check the contents of the package, type sudo dpkg-deb -c libprecice2_2.2.1_bionic.deb 
+    * To read installed contents of package, type sudo dpkg -L libprecice2_2.2.1_bionic.deb (???, why it says it is not installed)
+    * To check installed packages, sudo apt list --installed, you will find libprecice2/now 2.2.1 amd64 [installed,local]
+
+# Important Links
+
+# Notes for next meeting
+
+@ https://github.com/precice/tutorials/tree/develop/partitioned-heat-conduction, within the read me file, the recatangular domain Omega = [0,2] x [0,1] how is that.
+Also for the analytical solution check 37ff the solution is not in page 37
+
+# Example Problem
+This is an example problem to qualify Data communication of data communication
+developped in the adapter, it is a simple 2D problem, a 2x2 square plate
+heated from the bottom at 5 degrees, while its boundary is kept constant
+from the top at 0 degrees, and the sides having zero heat flux.
+
+# Usage
+There is HeatedPlate.geo file this the file defining the geometry and the 
+mesh generator using gmsh, it definies the 2x2 Plate and its 3 boundaries
+it can be open using gmsh,just make sure that gmsh is installed and 
+run at the current directory gmsh HeatedPlate.geo . The mesh is just a uniform
+recatangular mesh, you can adjust its size but changing the number of points
+on the lines even from the script of text editor.
+
+Also the mesh already saved from gmsh under HeatedPlate.msh, all you have to do
+is to run ElmerGrid 14 2 HeatedPlate.msh to generate mesh files necessary for
+Elmer Solver, these files will be generated in sub directory with name HeatedPlate. After that just run ElmerSolver case.sif. After the solver finishes
+a directory with the name Results will be created, inside it the .vtu file
+can be found and visualized in paraview.
+
+# For Compiling user defined solver with precice
+elmerf90 -o Coupler_Solver Coupler_Solver.F90 /usr/lib/x86_64-linux-gnu/libprecice.so.2
```

Comment:
I understand that `elmerf90` is a compiler wrapper that probably sets up the paths to the Elmer libraries and include files. Therefore, I assume that I can just assume `gfortran` instead of `elmerf90`.

The usual and "proper" way to link libraries when compiling with gfortran is using the `-I`, `-L`, and `-l` flags, as @BenjaminRodenberg has it here. Note that this is still missing the `-L` flag for the library path and it will only work on a system that already knows where `libprecice.so` is located.

In other adapters and [in the Fortran module](https://github.com/precice/fortran-module/blob/develop/examples/solverdummy/Makefile), we use `pkg-config` to get the right flags from the system: `$(shell pkg-config --libs libprecice)`. I would include this command, instead.

> I think this requires some explanation in order to know how to find the right path. https://precice.org/installation-linking.html should be a good starting point (yes/no @MakisH).

This is indeed the right place to link to. But, even better, I would create a `Makefile` or a `CMakeLists.txt`.

General comment: I see that this PR needs a lot of work to clean-up before it is in a state that can be merged. Don't give up: you already know how to do everything now and it would be really nice to manage to get this merged at the end! :smiley: Try to migrate your notes to small, easy-to-do issues and then it will start flowing a bit more easily.

Reply:
I added more clarification to the readme about the compilation of the adapter, and stated explicitly that it is not needed in the main directory, and the user has to refer to the readme of each problem. Regarding the compilation, using gfortran is an option but it is recommended to use the wrapper, because it is designed to make sure the compilation is compatible with Elmer, so the wrapper compile the code with same settings and same compiler version used for Elmer compilation. For linking preCICE,  the path of preCICE shared library is added to the compilation command as follows `elmerf90 -o Coupler_Solver Coupler_Solver.F90 /usr/lib/x86_64-linux-gnu/libprecice.so.2`.